packages/issuer/contracts/AbstractIssuer.sol

```diff
@@ -55,6 +57,14 @@ contract AbstractIssuer is Initializable, Ownable {
     }
 
     function requestIssueFor(bytes memory _data, address _owner) public returns (uint) {
+        (uint256 from, uint256 to, string memory deviceId) = decodeIssue(_data);
+        require(to > from, "Generation period invalid. 'to' is lower than 'from'");
+        require(to <= block.timestamp, "Generation period invalid. 'to' is higher than now");
+        require(
+            from >= deviceLatestGenerationTimestamp[deviceId],
```

Comment:
I don't think we want to have it sequential, user should be able to request certificates for arbitrary time range.